### PR TITLE
Make the mappings file into a proper component

### LINKED ISSUE #2166: Make the mappings file into a proper component
The file `src/ontology/components/mappings.owl` contains cross-references linking Uberon terms to corresponding terms in foreign ontologies (only FBbt for now). Those cross-references are merged into the main ontology by a custom logic (in `uberon.Makefile`).

That file needs to be declared as a proper “component” in the ODK sense so that it is merged using the standard ODK rules.

See #2163 for related discussion.

### PR DESCRIPTION
**Pull request by anitacaron**
Fixes #2166

### CODE REVIEW

File: src/ontology/uberon.Makefile [hunk 1]
```diff
@@ -1170,7 +1170,7 @@ $(TMPDIR)/uberon-taxmod-%.owl: ext.owl
 # ----------------------------------------
 
 # Generate cross-references from the FBbt mapping file
-$(COMPONENTSDIR)/mappings.owl: mappings/fbbt-mappings.sssom.tsv ../scripts/sssom2xrefs.awk
+$(OTHER_SRC): mappings/fbbt-mappings.sssom.tsv ../scripts/sssom2xrefs.awk
```
Comment (gouttegd) [inline src/ontology/uberon.Makefile:1173]: This is not future-proof. This works for now while `mappings.owl` is the only component, but won‘t work anymore if/when we add another component in the future.

I think it’s fine here to leave that rule as it were; that is, keep referring to the mappings component directly by its filename, not through the `OTHER_SRC` variable.

File: src/ontology/uberon.Makefile [hunk 2]
```diff
@@ -90,9 +90,9 @@ checks: $(REPORTDIR)/uberon-edit-xp-check $(REPORTDIR)/uberon-edit-obscheck.txt
 # TODO issues/contributor.owl not being updated atm.
 # TODO: for the seeds to be correctly imported, we probably need to merge phenoscape in here
 # TODO: Huge number of printouts that pollute the general logs
-$(OWLSRC): $(SRC) $(COMPONENTSDIR)/disjoint_union_over.ofn $(REPORTDIR)/$(SRC)-gocheck $(REPORTDIR)/$(SRC)-iconv $(SCRIPTSDIR)/expand-dbxref-literals.pl
+$(OWLSRC): $(SRCMERGED) $(COMPONENTSDIR)/disjoint_union_over.ofn $(REPORTDIR)/$(SRC)-gocheck $(REPORTDIR)/$(SRC)-iconv $(SCRIPTSDIR)/expand-dbxref-literals.pl
 	echo "STRONG WARNING: issues/contributor.owl needs to be manually updated."
-	$(OWLTOOLS) --no-logging $< $(COMPONENTSDIR)/disjoint_union_over.ofn issues/contributor.owl --merge-support-ontologies --merge-import http://purl.obolibrary.org/obo/uberon/components/mappings.owl --expand-macros -o  $@ &&  $(SCRIPTSDIR)/expand-dbxref-literals.pl $@ > $@.tmp
```
[the remark this answers is not in the document]
Comment (matentzn) [reply src/ontology/uberon.Makefile:95]: Double chec SRCMERGED has OTHER_SRC dependency

[the remark this answers is not in the document]
Comment (gouttegd) [reply src/ontology/uberon.Makefile:95]: @matentzn Surprisingly, `SRCMERGED` is _not_ dependent on `OTHER_SRC` in the standard, ODK-generated Makefile. This is something that should ultimately be fixed in the ODK itself.

Comment (gouttegd) [inline src/ontology/uberon.Makefile:95]: The `--merge-support-ontologies` has nothing to do with the subsequent `--merge-import` and is needed here to make sure the `disjoint_union_over.ofn` and `contributor.owl` are merged.

**Review by gouttegd — CHANGES_REQUESTED** [review]
To change:

- [ ] Add back the `--merge-support-ontologies` in `uberon.Makefile` line 95;
- [ ] Re-replace `$(OTHER_SRC)` by `$(COMPONENTSDIR)/mappings.owl` in `uberon.Makefile` line 1173

**Review by gouttegd — APPROVED** [review]

**Pull request merged**